share/dkg/dkg.go

```diff
@@ -614,13 +614,15 @@ func (d *DistKeyGenerator) ProcessResponses(bundles []*ResponseBundle) (*Result,
 		}
 	}
 
-	if !foundComplaint {
-		// there is no complaint !
-		if d.canReceive && d.statuses.CompleteSuccess() {
+	// there is no complaint in the responses received and the status matrix
+	// is all filled with success that means we can finish the protocol -
+	// regardless of the mode chosen (fast sync or not).
+	if !foundComplaint && d.statuses.CompleteSuccess() {
+		d.state = FinishPhase
```

Comment:
can we make sure we're only transitioning from response phase to finish phase here? just make sure there's no other possible state transition someone could sneak in some how

Reply:
Mhhh what do you have in mind ?
Here under this specific condition i directly put the phase to the finish line - once it's there no other functions can be called on the DKG.

Reply:
```
if d.State != responding {
  error
}
```
?

Reply:
I believe this is already taken care of at the top level of the function: https://github.com/drand/kyber/blob/7f0691e40632ada294ac4ce33c1b6a84ede1585b/share/dkg/dkg.go#L544

Reply:
throughout protocol, we only transition through the transition function that validates it's an allowed state transition. can we alternatively use the expected transition function here as well?